src/Persistence/Sql/Oracle/Query.php

```diff
@@ -4,8 +4,24 @@
 
 namespace Atk4\Data\Persistence\Sql\Oracle;
 
-class Query extends AbstractQuery
+use Atk4\Data\Persistence\Sql\Query as BaseQuery;
+
+class Query extends BaseQuery
 {
+    public function render(): string
+    {
+        if ($this->mode === 'select' && $this->main_table === null) {
+            $this->table('DUAL');
+        }
+
+        return parent::render();
+    }
+
+    public function groupConcat($field, string $delimiter = ',')
+    {
+        return $this->expr('listagg({field}, []) within group (order by {field})', ['field' => $field, $delimiter]);
```

Comment:
Which is lowest Oracle version which supports LISTAGG?

Reply:
11g2 https://www.toolbox.com/tech/oracle/blogs/oracle-sql-listagg-in-oracle-9i-022118/

I belive we do not use it Model/Ref thus this version is not strictly required.